src/main/scala/li/cil/oc/integration/appeng/NetworkControl.scala

```diff
@@ -221,17 +221,26 @@ object NetworkControl {
       val status = new CraftingStatus()
       Future {
         try {
-          val job = future.get() // Make 100% sure we wait for this outside the scheduled closure.
-          EventHandler.scheduleServer(() => {
-            val link = craftingGrid.submitJob(job, Craftable.this, cpu, prioritizePower, source)
-            if (link != null) {
-              status.setLink(link)
-              links += link
-            }
-            else {
-              status.fail("missing resources?")
-            }
-          })
+          while (!future.isDone) {
+            Thread.sleep(10)
```

Comment:
This will probably fail with "too long without yielding" (in case of long enough simulations) because any "client" code should fit in one tick timeslot

Reply:
This does not happen because the future gets executed async by a scala threadpool and the function returns the crafting status object. Thats also the reason you want the two new functions so you can track the status of the crafting job. The previous code also worked like this but directly called get which then caused problems because it was simulated twice at the same time. I have also already tested and used this and it works

Reply:
Oh, I didn't notice this code is within the Future clause itself, ok then.